clang/test/3C/arrbothmulti1.c

```diff
@@ -110,26 +110,26 @@ static int *mul2(int *x) {
 }
 
 int * sus(int *, int *);
-	//CHECK_NOALL: int * sus(int *, _Ptr<int> y);
-	//CHECK_ALL: _Array_ptr<int> sus(int *, _Ptr<int> y);
+	//CHECK_NOALL: int *sus(int *x : itype(_Ptr<int>), _Ptr<int> y) : itype(_Ptr<int>);
+	//CHECK_ALL: _Array_ptr<int> sus(int *x : itype(_Ptr<int>), _Ptr<int> y);
 
 int * foo() {
-	//CHECK_NOALL: int * foo(void) {
+	//CHECK_NOALL: _Ptr<int> foo(void) {
 	//CHECK_ALL: _Array_ptr<int> foo(void) {
         int * x = malloc(sizeof(int));
-	//CHECK: int * x = malloc<int>(sizeof(int));
+	//CHECK: _Ptr<int> x = malloc<int>(sizeof(int));
         int * y = malloc(sizeof(int));
 	//CHECK: _Ptr<int> y = malloc<int>(sizeof(int));
         int * z = sus(x, y);
```

Comment:
I guess what's happening here is that `sus` is no longer polluting things passed to it, or returned from it, so that's allowing `foo` to have a checked return type?